cdap-app-fabric/src/test/java/io/cdap/cdap/internal/app/program/MessagingProgramStatePublisherTest.java

```diff
@@ -76,7 +76,7 @@ public void testMultipleTopics() throws TopicNotFoundException, IOException {
     Map<String, String> properties = ImmutableMap.of(
       ProgramOptionConstants.PROGRAM_RUN_ID, GSON.toJson(runId));
     publisher.publish(Notification.Type.PROGRAM_STATUS, properties);
-    int topicNum = Math.abs(RUN_ID.hashCode() % NUM_PARTITIONS);
+    int topicNum = Math.abs(runId.getParent().getParent().hashCode()) % NUM_PARTITIONS;
```

Comment:
Please use runId.getApplication, it's more clear to reader

Reply:
AFAIK, runId.getApplication does not include version and namespace. It might be still ok for being used in partitioning.

Reply:
I think it's better to use ApplicationId than the name, I'll just have a separate variable for readability

Reply:
Oh, makes sense.